### PR TITLE
Add `TAB-INTP` compu method

### PR DESCRIPTION
**Pull request by KLBa**
Fixes [#12](https://github.com/mercedes-benz/odxtools/issues/12)

Katrin Bauer <[katrin.b.bauer@mercedes-benz.com](mailto:katrin.b.bauer@mercedes-benz.com)>, on behalf of [MBition GmbH](https://mbition.io/).
[Provider Information](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md)

### CODE REVIEW

File: odxtools/compumethods/readcompumethod.py [hunk 1]
```diff
@@ -168,7 +168,23 @@ def read_compu_method_from_odx(et_element, internal_type: DataType, physical_typ
...
-        return TabIntpCompuMethod(internal_type=internal_type, physical_type=physical_type)
+
+        scales = et_element.findall(
+            "COMPU-INTERNAL-TO-PHYS/COMPU-SCALES/COMPU-SCALE")
+        internal = [scale.findtext("LOWER-LIMIT") for scale in scales]
+        physical = [scale.findtext("COMPU-CONST/V") for scale in scales]
+
+        # cast to correct type
+        if internal_type in [DataType.A_INT32, DataType.A_UINT32]:
+            internal = [int(x) for x in internal]
```
Comment (andlaus) [inline odxtools/compumethods/readcompumethod.py:179]: shouldn't this better be done using `internal_type.from_string(x)`? Maybe there is an issue with this...

File: odxtools/compumethods/readcompumethod.py [hunk 2]
```diff
@@ -168,7 +168,23 @@ def read_compu_method_from_odx(et_element, internal_type: DataType, physical_typ
...
+        else:
+            internal = [float(x) for x in internal]
+        if physical_type in [DataType.A_INT32, DataType.A_UINT32]:
+            physical = [int(x) for x in physical]
```
Comment (andlaus) [inline odxtools/compumethods/readcompumethod.py:183]: same with `physical_type`?

File: odxtools/compumethods/tabintpcompumethod.py [hunk 3]
```diff
@@ -2,28 +2,144 @@
...
+                                      x: Union[int, float],
+                                      points: List[Tuple[Union[int, float], Union[int, float]]]) \
+            -> Union[float, int, None]:
+        for ((x0, y0), (x1, y1)) in zip(points[:-1], points[1:]):
+            if x0 <= x and x <= x1:
+                return y0 + (x - x0) * (y1 - y0) / (x1 - x0)
+
+        return None
+
+    def convert_physical_to_internal(self, physical_value: Union[int, float]):
```
Comment (andlaus) [inline odxtools/compumethods/tabintpcompumethod.py:119]: no type annotation for the return value. intentional? (same applies to a few other new functions.)

**Review by andlaus — COMMENTED** [review]
nice! My comments are just a few nits...

File: odxtools/compumethods/readcompumethod.py (hunk 1 above)
Comment (KLBa) [reply odxtools/compumethods/readcompumethod.py]: You're right. I maybe might have just forgotten about it...

File: odxtools/compumethods/tabintpcompumethod.py (hunk 3 above)
Comment (KLBa) [reply odxtools/compumethods/tabintpcompumethod.py]: Just me being sloppy. I added the type annotations.

**Comment by andlaus** [comment]
nice! thanks a lot!

**Pull request merged**